libtenzir/src/detail/loader_saver_resolver.cpp

```diff
@@ -123,6 +123,18 @@ struct try_plugin_by_uri_result {
   }
 };
 
+auto plugin_name_for_scheme(std::string_view scheme) -> std::string_view {
```

Comment:
Should this be a property of the loader and saver plugins?

Reply:
Maybe? On an architectural/conceptual level, it might make more sense that way. I'll try out what it'd look like.

Reply:
It might be better that way. I made the mapping 1-to-1: each plugin can support only a single URI scheme, which defaults to `plugin::name()`.